pseudomonas_analysis/2_identify_generic_genes_pathways_pao1.ipynb

```diff
@@ -0,0 +1,1764 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Note: We are manually changing the function to suit our nex experiment where we \n",
+    "# subset our training compendium. Ponyo will be updated to address this change to\n",
+    "# be more robust to user inputs\n",
+    "import glob\n",
+    "import warnings\n",
+    "from keras.models import load_model\n",
+    "from sklearn import preprocessing\n",
+    "\n",
+    "def get_sample_ids(experiment_id, dataset_name, sample_id_colname):\n",
+    "    \"\"\"\n",
+    "    Returns sample ids (found in gene expression df) associated with\n",
+    "    a given list of experiment ids (found in the metadata)\n",
+    "\n",
+    "    Arguments\n",
+    "    ----------\n",
+    "    experiment_ids_file: str\n",
+    "        File containing all cleaned experiment ids\n",
+    "\n",
+    "    dataset_name: str\n",
+    "        Name for analysis directory. Either \"Human\" or \"Pseudomonas\"\n",
+    "\n",
+    "    sample_id_colname: str\n",
+    "        Column header that contains sample id that maps expression data\n",
+    "        and metadata\n",
+    "\n",
+    "    \"\"\"\n",
+    "    base_dir = os.path.abspath(os.path.join(os.getcwd(), \"../\"))\n",
+    "\n",
+    "    if \"pseudomonas\" in dataset_name.lower():\n",
+    "        # metadata file\n",
+    "        mapping_file = os.path.join(\n",
+    "            base_dir, dataset_name, \"data\", \"metadata\", \"sample_annotations.tsv\"\n",
+    "        )\n",
+    "\n",
+    "        # Read in metadata\n",
+    "        metadata = pd.read_csv(mapping_file, header=0, sep=\"\\t\", index_col=0)\n",
+    "\n",
+    "        selected_metadata = metadata.loc[experiment_id]\n",
+    "        sample_ids = list(selected_metadata[sample_id_colname])\n",
+    "\n",
+    "    else:\n",
+    "        # metadata file\n",
+    "        mapping_file = os.path.join(\n",
+    "            base_dir, dataset_name, \"data\", \"metadata\", \"recount2_metadata.tsv\"\n",
+    "        )\n",
+    "\n",
+    "        # Read in metadata\n",
+    "        metadata = pd.read_csv(mapping_file, header=0, sep=\"\\t\", index_col=0)\n",
+    "\n",
+    "        selected_metadata = metadata.loc[experiment_id]\n",
+    "        sample_ids = list(selected_metadata[sample_id_colname])\n",
+    "\n",
+    "    return sample_ids\n",
+    "\n",
+    "def shift_template_experiment_tmp(\n",
+    "    normalized_data_file,\n",
+    "    selected_experiment_id,\n",
+    "    sample_id_colname,\n",
+    "    NN_architecture,\n",
+    "    dataset_name,\n",
+    "    scaler,\n",
+    "    local_dir,\n",
+    "    base_dir,\n",
+    "    run,\n",
+    "):\n",
+    "    \"\"\"\n",
+    "    Generate new simulated experiment using the selected_experiment_id as a template\n",
+    "    experiment using the same workflow as `simulate_by_latent_transform`\n",
```

Comment:
What is `simulate_by_latent_transform`? It may help readers to reference what file it's in

Reply:
Ah. Since I copied this function, its no longer in the same file as the `simulate_by_latent_transform.py` function that includes a description. I'll update this text